test/front_desk_test.rb

```diff
@@ -0,0 +1,374 @@
+require_relative 'test_helper'
+describe "front_desk" do 
+  def hotel_manager 
+    return Hotel::FrontDesk.new()
+  end 
+  describe "#reservation_cost" do 
+    it "calculate cost for a reservation" do 
+      manager = hotel_manager
+      reservation_1 = Hotel::Reservation.new(
+        date_range: Hotel::DateRange.new(
+          start_date:[2020,3,2],end_date:[2020,3,5]
+        )
+      )
+      reservation_2 = Hotel::Reservation.new(
+        date_range: Hotel::DateRange.new(
+          start_date:[2020,4,1],end_date:[2020,4,8]
+        )
+      )
+  
+      manager.reservations << reservation_1
+      manager.reservations << reservation_2
+      cost = manager.reservation_cost(reservation_1)
+      expect(cost).must_equal 600
+      cost2 = manager.reservation_cost(reservation_2)
+      expect(cost2).must_equal 1400
+    end
+    it "calculate cost for block individual room as well" do 
+      manager = hotel_manager
+      start_date = [2020,4,9]
+      end_date = [2020,4,11]
+      manager.create_block(start_date,end_date,5,180)
+      block_id = manager.reservations[0].reservation_id.to_s[0..4]
+      reservation = manager.book_room_of_block(block_id)
+
+      cost_of_block_room = manager.reservation_cost(reservation)
+      expect(cost_of_block_room).must_equal 360
+    end 
+
+  end  
+
+  describe "#list_all" do
+    it "return a list of all reservations" do 
+      manager = hotel_manager
+      reservation_1 = Hotel::Reservation.new(
+        reservation_id:1234,
+        date_range: Hotel::DateRange.new(
+          start_date:[2020,3,2],end_date:[2020,3,5]
+        )
+      )
+      reservation_2 = Hotel::Reservation.new(
+        reservation_id:1235,
+        date_range: Hotel::DateRange.new(
+          start_date:[2020,4,1],end_date:[2020,4,8]
+        )
+      )
+      manager.reservations << reservation_1
+      manager.reservations << reservation_2
+      result = manager.list_all
+      expect(result).must_match (/2020-03-02/)
+      expect(result).must_match (/2020-03-05/)
+      expect(result).must_match (/2020-04-01/)
+      expect(result).must_match (/2020-04-08/)
+      expect(result).wont_match (/1999-04-08/)
+      expect(result).must_match (/1234/)
+      expect(result).must_match (/1235/)
```

Comment:
Much of why we like to just return data structures directly from things in `lib` is that we want to be able to unit-test them using things like `must_equal` without having to rely on regular expressions.  (Not that regular expressions aren't awesome.  😉 )